src/components/CustomCollection/EditableCollectionTitle.test.tsx

```diff
@@ -15,231 +15,76 @@ describe('EditableCollectionTitle component', () => {
-      const handleOnSave = jest.fn()
-      render(
-        <EditableCollectionTitle
-          collectionId="testCollection123"
-          text="Test Collection"
-          onSave={handleOnSave}
-          onDelete={jest.fn()}
-        />
-      )
-      const editTitle = screen.getByRole('button', {
-        name: 'Edit Test Collection collection title',
-      })
-      userEvent.click(editTitle)
-      const input = screen.getByRole('textbox')
-      userEvent.click(input)
-      userEvent.clear(input)
-      userEvent.type(input, 'Updated Title')
-      userEvent.tab()
-      expect(handleOnSave).toHaveBeenCalled()
-      expect(
-        screen.getByRole('button', {
-          name: 'Edit Updated Title collection title',
-        })
-      ).toHaveTextContent('Updated Title')
-    })
-  })
+      screen.getByRole('textbox', { name: 'Collection Title' })
+    ).toBeInTheDocument()
 
-  it('reverts back to the previous value if the input is cleared', () => {
-    const handleOnSave = jest.fn()
+    expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument()
 
-    render(
-      <EditableCollectionTitle
-        collectionId="testCollection123"
-        text="Test Collection"
-        onSave={handleOnSave}
-        onDelete={jest.fn()}
-      />
-    )
-    const editTitle = screen.getByRole('button', {
-      name: 'Edit Test Collection collection title',
-    })
-    userEvent.click(editTitle)
-    const input = screen.getByRole('textbox')
-    userEvent.click(input)
-    userEvent.clear(input)
-    fireEvent.blur(input)
-    expect(handleOnSave).not.toHaveBeenCalled()
     expect(
-      screen.getByRole('button', {
-        name: 'Edit Test Collection collection title',
-      })
-    ).toHaveTextContent('Test Collection')
+      screen.getByRole('button', { name: 'Save name' })
+    ).toBeInTheDocument()
   })
+})
 
-  it('deletes the collection if the input is left empty and there is no existing value', () => {
-    const handleOnSave = jest.fn()
-    const handleOnDelete = jest.fn()
-
-    render(
-      <EditableCollectionTitle
-        collectionId="testCollection123"
-        text=""
-        onSave={handleOnSave}
-        onDelete={handleOnDelete}
-      />
-    )
+it('saves the form', async () => {
+  const mockHandleOnSave = jest.fn()
+  render(
+    <EditableCollectionTitle
+      collectionId="testCollection123"
+      text="Test Collection"
+      onSave={mockHandleOnSave}
+      onCancel={jest.fn()}
+      onDelete={jest.fn()}
+      isEditing={true}
+    />
+  )
+  const input = screen.getByRole('textbox', { name: 'Collection Title' })
+  userEvent.type(input, 'New Title')
+
+  const saveNameButton = screen.getByRole('button', { name: 'Save name' })
+  userEvent.click(saveNameButton)
+  // #TODO "errors" when clicking submit on form; expected behavior by jsdom
+  // <<Error: Not implemented: HTMLFormElement.prototype.submit>>
+  // Should I ignore the error?
+  expect(mockHandleOnSave).toBeCalledTimes(1)
```

Comment:
not a blocker but you can also use `toBeCalledWith('New Title')` to test that the handler was called with the expected param(s)